api/src/main/java/org/terrakube/api/plugin/streaming/StreamingConfiguration.java

```diff
@@ -1,27 +1,81 @@
+import java.security.KeyStore;
+import java.security.SecureRandom;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.data.redis.connection.RedisStandaloneConfiguration;
+import org.springframework.data.redis.connection.jedis.JedisClientConfiguration;
 import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
 import org.springframework.data.redis.core.RedisTemplate;
 
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
 @Configuration
 public class StreamingConfiguration {
 
     @Bean
-    public RedisTemplate<String, Object> redisTemplate(JedisConnectionFactory jedisConnectionFactory) {
-        RedisTemplate<String, Object> template = new RedisTemplate<>();
-        template.setConnectionFactory(jedisConnectionFactory);
-        return template;
+    SSLSocketFactory sslSocketFactory(StreamingProperties props) throws Exception {
+        if (!props.isSsl()) {
+            return SSLContext.getDefault().getSocketFactory();
+        }
+        KeyStore trustStore = KeyStore.getInstance("jks");
+        InputStream inputStream = null;
+        try {
+            inputStream = new FileInputStream(props.getTruststorePath());
+            trustStore.load(inputStream, props.getTruststorePassword().toCharArray());
+        } catch (Exception e) {
+            log.error(e.getMessage());
+        } finally {
+            if (inputStream != null) {
+                inputStream.close();
+            }
+        }
+
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("PKIX");
+        trustManagerFactory.init(trustStore);
+        TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
+
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, trustManagers, new SecureRandom());
+        return sslContext.getSocketFactory();
     }
+
     @Bean
-    JedisConnectionFactory jedisConnectionFactory(StreamingProperties streamingProperties) {
-        RedisStandaloneConfiguration redisStandaloneConfiguration = new RedisStandaloneConfiguration(streamingProperties.getHostname(), streamingProperties.getPort());
-        redisStandaloneConfiguration.setPassword(streamingProperties.getPassword());
-        JedisConnectionFactory jedisConFactory = new JedisConnectionFactory(redisStandaloneConfiguration);
+    JedisConnectionFactory jedisConnectionFactory(StreamingProperties props, SSLSocketFactory sslSocketFactory) {
+        RedisStandaloneConfiguration redisStandaloneConfiguration = new RedisStandaloneConfiguration(
+                props.getHostname(), props.getPort());
+        redisStandaloneConfiguration.setPassword(props.getPassword());
+        redisStandaloneConfiguration.setUsername(props.getUsername());
```

Comment:
@alfespa17 @rogerxaic Won't this change cause a problem with the default Redis deployment because it means a username will be specified all the time and cause the `RedisHealthIndicator` to fail? This is in relation to this: https://github.com/AzBuilder/terrakube/pull/713#issuecomment-1961261563